tests/erlang_tests/is_record.erl

```diff
@@ -0,0 +1,48 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2024 Tomasz Sobkiewicz <tomasz.sobkiewt>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(is_record).
+
+-record(person, {id, name, age}).
+
+-export([start/0, id/1]).
+
+-define(ID(Arg), ?MODULE:id(Arg)).
+
+start() ->
+    Person = ?ID(person),
+    Foo = ?ID(foo),
+    Mike =
+        #person{id = 1,
+                name = "Mike",
+                age = 32},
+    true = erlang:is_record({person, 1, 2, 3}, Person),
```

Comment:
This should be enough, however let's future-proof it more: let's use id also for `{person, 1, 2, 3}`. Such as:
erlang:is_record(ID({person, 1, 2, 3}), Person).